source/upgrade.txt

```diff
@@ -105,16 +88,121 @@ Version 5.0 Breaking Changes
+  reading from a secondary member of a replica set, use the ``isSecondaryOk()`` methods from
+  these classes instead.
+
+- Removes the ``DBCollection.getStats()`` and ``DBCollection.isCapped()``
+  helper methods for the ``collStats`` command. Instead of these methods, you can use the
+  ``$collStats`` aggregation pipeline stage. For an example of how to use this pipeline
+  stage, see :manual:`What's New <java-deprecations-4.11>` for v4.11 of the {+driver-short+}.
+
+- Removes the ``IterableCodec`` and ``MapCodec`` types.
+  Use ``MapCodecProvider`` instead of ``MapCodec``, and ``CollectionCodecProvider``
+  instead of ``IterableCodec``.
+
+- Removes the ``sharded`` and ``nonAtomic`` options from the
+  ``MapReducePublisher`` and ``MapReduceIterable`` classes.
 
-- This driver version removes the
+- Removes the following methods for use with ``geoHaystack`` indexes:
+
+  - ``Indexes.geoHaystack()``
+  - ``IndexOptions.getBucketSize()``
+  - ``IndexOptions.bucketSize()``
+  
+  Instead, you can use the ``$geoNear`` aggregation pipeline stage or a geospatial
+  query operator on a 2d index. For more information, see the
+  :manual:`Geospatial Queries page </geospatial-queries>` in the {+mdb-server+} manual.
+
+- Removes the ``oplogReplay`` option from find operations. This
+  includes the following methods:
+
+  - ``DBCursor.oplogReplay()``
+  - ``DBCollectionFindOptions.isOplogReplay()``
+  - ``DBCollectionFindOptions.oplogReplay()``
+  - ``FindPublisher.oplogReplay()``
+  - ``FindIterable.oplogReplay()``
+
+- Removes the following ``Exception`` constructors:
+
+  - ``MongoBulkWriteException(BulkWriteResult, List<BulkWriteError>, WriteConcernError, ServerAddress)``
+  - ``MongoCursorNotFoundException(long, ServerAddress)``
+  - ``MongoQueryException(ServerAddress, int, String)``
+  - ``MongoQueryException(ServerAddress, int, String, String)``
+  - ``MongoQueryException(MongoCommandException)``
+
+- Removes the following overloads for the ``BulkWriteResult.acknowledged()`` method:
+
+  - ``acknowledged(Type, int, List<BulkWriteUpsert>)``
+  - ``acknowledged(Type, int, Integer, List<BulkWriteUpsert>)``
+  - ``acknowledged(int, int, int, Integer, List<BulkWriteUpsert>)<String>)``
+
+- Removes the following ``ChangeStreamDocument`` constructors:
+
+  - ``ChangeStreamDocument(String, BsonDocument, BsonDocument, BsonDocument, TDocument, TDocument, BsonDocument, ...)``
+  - ``ChangeStreamDocument(String, BsonDocument, BsonDocument, BsonDocument, TDocument, BsonDocument, BsonTimestamp, ...)``
+  - ``ChangeStreamDocument(OperationType, BsonDocument, BsonDocument, BsonDocument, TDocument, BsonDocument, BsonTimestamp, ...)``
+
+- Removes the following constructors for events:
+
+  - ``CommandEvent(RequestContext, int, ConnectionDescription, String)``
+  - ``CommandEvent(int, ConnectionDescription, String)``
+  - ``CommandEvent(RequestContext, long, int, ConnectionDescription, String)``
```

Comment:
FYI for tech reviewer: the [ticket for this change](https://jira.mongodb.org/browse/DOCSP-34140) lists only one `CommandEvent` constructor removed, but it looks to me like [the PR actually removed three.](https://github.com/mongodb/mongo-java-driver/pull/1268/commits/20bad8b45eccaca5db5a7cd3e71256edbdc52267#diff-bc9f75fc6edf35fd20add96a4c510b7495dc96acb52106aae15e663a8ea85286) Please let me know if this is incorrect